var/spack/repos/builtin/packages/pango/package.py

```diff
@@ -35,13 +35,18 @@ class Pango(Package):
     list_url = "http://ftp.gnome.org/pub/gnome/sources/pango/"
     list_depth = 2
 
-    version('1.36.8', '217a9a753006275215fa9fa127760ece')
+    version('1.40.3', '17c26720f5a862a12f7e1745e2f1d966')
     version('1.40.1', '6fc88c6529890d6c8e03074d57a3eceb')
+    version('1.36.8', '217a9a753006275215fa9fa127760ece')
 
     depends_on("pkg-config", type="build")
     depends_on("harfbuzz")
     depends_on("cairo")
     depends_on("glib")
+    
+    def url_for_version(self, version):
+        base_url = "http://ftp.gnome.org/pub/gnome/sources/pango"
+        return "{}/{}/pango-{}.tar.xz".format(base_url , version.up_to(2), version)
```

Comment:
The reason the tests are failing is because this particular syntax is only supported in Python 2.7+, not 2.6. You'll need to use `{0}...{1}...{2}` or `{0}...{1.up_to(2)}...{1}` instead of `{}...{}...{}`.